### PR TITLE
Replacing pyfits with Astropy to read FITS

### LINKED ISSUE #3002: Use astropy instead of pyfits to read FITS images
pyfits is not currently installable on 3.5, and it looks like AstroPy is more actively maintained.

### PR DESCRIPTION
**Pull request by alexandrejaguar**
## Description

Closes #3002.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [X] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/io/tests/test_fits.py [hunk 1]
```diff
@@ -5,53 +5,50 @@
 from skimage._shared import testing
 
 
-pyfits_available = True
```
Comment (hmaarrfk) [inline skimage/io/tests/test_fits.py]: Can probably be
```
pytest.importorskip("astropy")
```

https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency

And most of `skipif` statements can be ignored.

Comment (jni) [reply skimage/io/tests/test_fits.py]: I like this, thanks for pointing it out, @hmaarrfk!

**Comment by emmanuelle** [comment]
I also like more the syntax with ``pytest.importorskip``. The fits format is quite domain-specific, what was the reason to include it in scikit-image (I thought we only supported cross-domain formats, but I might have missed something).

**Comment by alexandrejaguar** [comment]
@jni @emmanuelle let's see if I can implement @hmaarrfk's idea!

**Comment by jni** [comment]
I don't understand why the tests appear to be passing rather than skipped...?

https://travis-ci.org/scikit-image/scikit-image/jobs/537512456#L2438

@hmaarrfk any ideas? (That build [doesn't have astropy installed](https://travis-ci.org/scikit-image/scikit-image/jobs/537512456#L989)

**Comment by stefanv** [comment]
On Sun, 26 May 2019 06:02:49 -0700, Emmanuelle Gouillart wrote:
> I also like more the syntax with ``pytest.importorskip``. The fits
> format is quite domain-specific, what was the reason to include it in
> scikit-image (I thought we only supported cross-domain formats, but I
> might have missed something).

Mainly because James Turner was at SciPy and because we thought back
then that `skimage.io` would be the entry portal for all data.  Over
time, we've settled on the "arrays as inputs" paradigm, and moved
further away from the plugin infrastructure in favor of a single image
reader solution (`imageio`).  Which supports FITS through a plugin :)

**Comment by hmaarrfk** [comment]
@jni I think it is because instead of collecting the tests, pytest raises an internal exception and skips the whole file before collection.

```python
$ ipython                                                                                                                                                                                                      
import pyPython 3.6.7 | packaged by conda-forge | (default, Feb 28 2019, 09:07:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pytest                                                                                                                                                                                          

In [2]: pytest.importorskip('astropy')                                                                                                                                                                         
---------------------------------------------------------------------------
Skipped                                   Traceback (most recent call last)
<ipython-input-2-2e443ca88ab1> in <module>
----> 1 pytest.importorskip('astropy')

~/miniconda3/envs/skdev/lib/python3.6/site-packages/_pytest/outcomes.py in importorskip(modname, minversion, reason)
    170         if reason is None:
    171             reason = "could not import %r" % (modname,)
--> 172         raise Skipped(reason, allow_module_level=True)
    173     mod = sys.modules[modname]
    174     if minversion is None:

Skipped: could not import 'astropy'
```

I'm kinda for the convenience as opposed to pretty-printing the tests.

**Comment by jni** [comment]
Wow, just tested this locally — indeed it doesn't print the skipped file, even in verbose mode. That's kinda crummy. =\

**Comment by hmaarrfk** [comment]
Yeah, for others  to see

```
$ pytest skimage/io/tests/test_fits.py                                                                                                                                                                         
Test session starts (platform: linux, Python 3.6.7, pytest 4.5.0, pytest-sugar 0.9.2)
rootdir: /home/mark2/git/scikit-image/skimage/io
plugins: xdist-1.28.0, sugar-0.9.2, localserver-0.5.0, forked-1.0.2
collecting ... 

Results (0.09s):
```

**Comment by sciunto** [comment]
@stefanv's remark would suggest to get ride of astropy in favor of imageio (if truely equivalent, tests/investigation needed) https://imageio.readthedocs.io/en/stable/formats.html?highlight=fits
and @emmanuelle's remark raise the question to maintain the support of the FITS format.

I'm OK to merge this PR and open a SKIEP issue with these two points.

**Comment by alexandrejaguar** [comment]
@sciunto that's fine by me too. I have several points to maintain FITS support, but let's wait for the SKIEP :)

**Comment by sciunto** [comment]
Let's SKIP this sort of details!

**Comment by hmaarrfk** [comment]
sigh, unfortunately, there are too many errors in the build system for me to want to pull this in with confidence.

**Comment by alexandrejaguar** [comment]
@hmaarrfk the major issue is this (3/4 "pythons"):
```
Warning, treated as error:
/home/travis/venv/lib/python3.7/site-packages/skimage/transform/pyramids.py:docstring of skimage.transform.pyramid_laplacian:71:duplicate citation R577, other instance in /home/travis/build/scikit-image/scikit-image/doc/source/api/skimage.util.rst
Makefile:53: recipe for target 'html' failed
make[1]: *** [html] Error 2
make[1]: Leaving directory '/home/travis/build/scikit-image/scikit-image/doc'
Makefile:26: recipe for target 'html' failed
make: *** [html] Error 2
The command "tools/travis/script.sh" exited with 2.
```

**Comment by jni** [comment]
iirc the "duplicate citation" errors can be fixed by clearing all the Travis caches.

**Comment by hmaarrfk** [comment]
I just cleared everything, lets see what happens.

**Comment by hmaarrfk** [comment]
imread bug seems to be bugging out.

**Comment by alexandrejaguar** [comment]
@hmaarrfk I wonder what we could do to solve that thing.

**Comment by hmaarrfk** [comment]
We would have to work with https://github.com/luispedro/imread/issues/35
to help them get real wheels uploaded.


Releasing wheels is seriously hard, too hard....

**Comment by hmaarrfk** [comment]
I think they might have built with newer OS, which makes things technically incompatible with the old OS we test with, that is my guess at least.

**Comment by hmaarrfk** [comment]
Quite frankly, i would rip out the whole `skimage.io` module.

**Comment by alexandrejaguar** [comment]
> Quite frankly, i would rip out the whole `skimage.io` module.

Don't do that. We love it!

**Comment by hmaarrfk** [comment]
maybe we can just express our opinions in https://github.com/scikit-image/scikit-image/issues/1605

**Pull request merged**